### ISSUE TITLE
support for different fonts (family, trait & size) in highlighter

### ISSUE THREAD

**Issue by NeomMob** [shysaur/Fragaria#92]
I didn't found how to set up differents fonts in the highlighter. Is it supported and if not, do you have any direction for the implementation?

**Comment by shysaur** [comment]
Currently, highlighting by changing anything but color is unsupported because, all highlighting is done by setting [temporary attributes](https://developer.apple.com/library/mac/documentation/Cocoa/Reference/ApplicationKit/Classes/NSLayoutManager_Class/#//apple_ref/occ/instm/NSLayoutManager/addTemporaryAttributes:forCharacterRange:) on the layout manager. Temporary attributes can't affect layout; this means that trying to set the font with temporary attributes won't have any effect.

The only way to do this would be to change the attributes in the text storage, but since a text view's text storage can be modified by other parts of the application in unexpected ways, this is not really feasible. Some trickery would be needed to hide Fragaria's attributes from the rest of the application. But, even if we do that, it might interact in some unexpected way with the text view, because Fragaria sets [isRichText](https://developer.apple.com/library/mac/documentation/Cocoa/Reference/ApplicationKit/Classes/NSTextView_Class/#//apple_ref/occ/instp/NSTextView/richText) to YES.

To sum it up, it's probably doable, but not in an easy way.

**Comment by shysaur** [comment]
BTW, this is a duplicate of #70.

**Comment by NeomMob** [comment]
Thanks for the explanation. I tried to modify if (mode from temporary attributes to directly setting attribute in the text storage) but performance is really impacted (+ the potential side effects you described)

**Comment by shysaur** [comment]
Fixed in 6b16cef9 (see issue #70).
